package.py

```diff
@@ -0,0 +1,162 @@
+    except FileNotFoundError:
+        print("Attempting clean up unnecessary. No temp_folder/ found at: ", dist_folder)
+
+def setup_environment():
+    requirements_results = subprocess.run(
+        ['pip','install','-r', 'requirements.txt'],
+        cwd=PARENT_DIR,
+        **subprocess_kwargs
+    )
+
+    if requirements_results.returncode:
+        log_build_error(build_logger,requirements_results)
+        sys.exit(requirements_results.returncode)
+        
+    print('Environment setup complete')
+        
+
+def build_backend(): 
+    print("Building the backend")
+    build_results = subprocess.run(
+        ['python','setup.py','bdist_wheel'],
+        cwd=PARENT_DIR, #only works locally
+        **subprocess_kwargs
+    )
+
+    if build_results.returncode:
+        log_build_error(build_logger,build_results)
+        sys.exit(build_results.returncode)
+    
+    expected_whl_path = PARENT_DIR/('dist/seeq_plot_curve-'+__version__+'-py3-none-any.whl')
+    if os.path.exists(expected_whl_path):
+        print(".whl generated at: ", expected_whl_path)
+    else:
+        error_message = "failed to generate: " + str(expected_whl_path)
+        raise FileNotFoundError(error_message) 
+    
+    print('Building backend complete')
+
+def generate_requirements_txt():
+    print("Generating backend requirements.txt file")
+    path = PARENT_DIR / "temp_folder/data-lab-functions/requirements.txt"
+    path.parent.mkdir(parents=True, exist_ok=True)
+    requirements = 'seeq_plot_curve-'+__version__+'-py3-none-any.whl'
+
+    with open(path, 'w') as f:
+        f.write(requirements)
+
+    if path.exists():
+        print("Successfully generated backend requirements.txt")
+    else:
+        print("Failed to generate backend requirements.txt")
+ 
+paths = [
+    [PARENT_DIR/('dist/seeq_plot_curve-'+__version__+'-py3-none-any.whl'), PARENT_DIR/(('temp_folder/data-lab-functions/seeq_plot_curve-'+__version__+'-py3-none-any.whl'))],
+    [PARENT_DIR/'seeq/addons/plot_curve/deployment_notebook/PlotCurve.ipynb',PARENT_DIR/'temp_folder/data-lab-functions/PlotCurve.ipynb'],
+    [PARENT_DIR/ 'addon.json',PARENT_DIR/'temp_folder/addon.json']
+]
+
+def move_artifacts():
+    build_logger.info("Moving artifacts into .addon directory:")
+    for artifact in paths:
+        print("Copying ,", artifact[0],"to ", artifact[1])
+        current_path = artifact[0]
+        new_path = artifact[1]
+        if current_path.is_dir():
+            shutil.copytree(str(current_path), str(new_path))
+        else:
+            new_path.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copy(str(current_path), str(new_path))
+
+        if new_path.exists():
+            build_logger.info(f"\t{current_path} moved successfully to {new_path}")
+        else:
+            build_logger.error(f'\t{current_path} failed to move to {new_path}')
+
+
+def zip_items():
+    zipName = "seeq-plot-curve-" + __version__ + ".addon"
```

Comment:
The `__version__` number in the filename doesn't match the version number of the Add-on. Ideally, they would match - and use the version number from the `addon.json` file - so_ anyone could just look at the filename and know what version it is.

We only allow one version of a particular Add-on to be in the gallery.

The Add-on Manager TeamCity build uses python to remove the previous version of the Add-on files in JFrog and replace them with the new version.

For Seeq ML, I believe @wizzle3d worked around the problem by removing the version number from the filename. Not ideal, but it worked. It would be great/best if we could figure out a way to configure https://teamcity.seeq-labs.com/admin/editRunType.html?id=buildType:AppliedResearch_UdfUiDeploymentExperimental&runnerId=RUNNER_29&cameFromUrl=%2Fadmin%2FeditBuildRunners.html%3Fid%3DbuildType%253AAppliedResearch_UdfUiDeploymentExperimental%26init%3D1&cameFromTitle= so it replaces the previous version even if the version number is different.

Reply:
I'll dig into this! Might not make this next push but potentially the next one

Reply:
I added in a step in the packaging that would ensure the addon contains the version in `__version__`!